src/test/java/de/sakpaas/backend/service/LocationApiSearchDasTest.java

```diff
@@ -0,0 +1,39 @@
+package de.sakpaas.backend.service;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import de.sakpaas.backend.HappyHamsterTest;
+import de.sakpaas.backend.dto.OsmResultLocationListDto;
+import de.sakpaas.backend.util.OsmImportConfiguration;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+class LocationApiSearchDasTest extends HappyHamsterTest {
+
+  final String country = "DE";
+  @Autowired
+  LocationApiSearchDas locationApiSearchDas;
+
+  @Test
+  void testEmptyShoptypeList() {
+    OsmImportConfiguration importConfiguration = new OsmImportConfiguration();
+
+
+    importConfiguration.setShoptypes(new ArrayList<>());
+    List<OsmResultLocationListDto.OsmResultLocationDto> results =
+        locationApiSearchDas.getLocationsForCountry(importConfiguration);
+
+    assertEquals(results, new ArrayList<>(), "Result List should be empty");
+
```

Comment:
Die Tests sind ja immer noch nicht ausführlicher geworden :P

Reply:
Tell me what to test ^^

Reply:
Du solltest auf jeden Fall `restTemplate.getForEntity()` Mocken, da du in einem Unit Test keine Verbindungen nach außen machen darfst.

Dann könntest du testen:
* ob alle angegebenen `shoptypes` auch wirklich in der Query landen
* ob das angegebene `country` auch in der Query vorhanden ist
* ob die ankommenden Locations auch zurück gegeben werden

Reply:
Das Rest-Template darfst du gerne mocken, wenn du dich dazu berufen fühlst ^^ (https://www.baeldung.com/spring-mock-rest-template)

Habs versucht, ich find aber, dass die Arbeit da in keinem Verhältnis zum Output steht. Dafür müsste man die Funktion noch ziemlich umbauen...

Reply:
Den Stringbuilder hab ich aber getestet... Er baut den String richtig :) xD

Reply:
Ich habs jz 3 Stunden versucht. Bekomme es nicht hin